### PR TITLE
[CDAP-20434] Add e2e test for pulling pipeline version from git

### PR DESCRIPTION
**Pull request by arjan-bal**
E2E Test for Pulling Pipeline version from Git

## Description
The test performs the following steps:
1. Create a pipeline
2. Push initial version to git.
3. Edit the pipeline and create a new version.
4. Pull the initial version back from Git.

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [x] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [CDAP-20434](https://cdap.atlassian.net/browse/CDAP-20434)

## Screenshots

[CDAP-20434]: https://cdap.atlassian.net/browse/CDAP-20434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/e2e-test/features/source.control.management.sync.apps.feature [hunk 1]
```diff
@@ -53,3 +53,35 @@ Feature: Source Control Management - Pulling and pushing applications
...
+    # Setup Repo config
+    When Open Source Control Management Page
+    When Initialize the repository config
+    When Deploy and test pipeline "pipeline_edit_test" with pipeline JSON file "logs_generator.json"
+    # Push version 1 of the pipeline to git.
+    Then Click push button in Actions dropdown
+    Then Commit changes with message "upload pipeline to Git"
+    Then Banner is shown with message "Successfully pushed pipeline pipeline_edit_test"
+    # Edit the pipeline to create a new LCM version.
+    Then Click edit button in Actions dropdown
```
Comment (samdgupi) [inline src/e2e-test/features/source.control.management.sync.apps.feature]: nit: We can have one step to create a new version.

File: src/e2e-test/features/source.control.management.sync.apps.feature [hunk 2]
```diff
@@ -53,3 +53,35 @@ Feature: Source Control Management - Pulling and pushing applications
...
+    Then Edit should redirect to studio page
+    Then Click on file source node property button
+    Then Modify reference name
+    Then Click deploy should succeed
+    Then Pipeline should show latest details
```
Comment (samdgupi) [inline src/e2e-test/features/source.control.management.sync.apps.feature]: nit: Prefer a named way to identify details e.g details1, details2

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java [hunk 3]
```diff
@@ -132,11 +133,19 @@ public void verifyUIDirectedToInitialPage() {
...
     ElementHelper.clickOnElement(Helper.locateElementByTestId("pipeline-actions-btn"));
     ElementHelper.clickOnElement(Helper.locateElementByTestId("push-pipeline"));
   }
 
+  @Then("Click pull button in Actions dropdown")
+  public void clickPullAction() {
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("pipeline-actions-btn"));
+    ElementHelper.clickOnElement(Helper.locateElementByTestId("pull-pipeline"));
+    Commands.waitForLoading();
+    WaitHelper.waitForPageToLoad();
```
Comment (samdgupi) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java:147]: why do we need two wait ?

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java [hunk 4]
```diff
@@ -146,8 +155,10 @@ public void commitPipeline(String message) {
 
   @Then("Banner is shown with message {string}")
   public void pipelineBannerIsShownWithMessage(String message) {
-    WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("alert"));
+    WebElement alert = Helper.locateElementByTestId("alert");
+    WaitHelper.waitForElementToBeDisplayed(alert);
     Assert.assertTrue(ElementHelper.getElementText(Helper.locateElementByTestId("alert")).contains(message));
+    WaitHelper.waitForElementToBeHidden(alert, 7);
```
Comment (samdgupi) [inline src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java]: nit: constant "ALERT_DISPLAY_TIMEOUT"

**Review by samdgupi — APPROVED** [review]

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java (hunk 3 above)
Comment (arjan-bal) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java:147]: The page refreshes. If we locate an element before the page has refreshed, the reference becomes invalid after the re-load.

**Review by de-lan — APPROVED** [review]

File: src/e2e-test/features/source.control.management.sync.apps.feature (hunk 2 above)
Comment (arjan-bal) [reply src/e2e-test/features/source.control.management.sync.apps.feature]: Done.

File: src/e2e-test/features/source.control.management.sync.apps.feature (hunk 1 above)
Comment (arjan-bal) [reply src/e2e-test/features/source.control.management.sync.apps.feature]: Done.

File: src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java (hunk 4 above)
Comment (arjan-bal) [reply src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SourceControlManagement.java]: Done.

**Pull request merged**